File: js/task1b/S1_IF_02.js
// S1_IF_02 - Score to grade

function grade(score) {
  if (score < 0 || score > 100) return null;

  if (score >= 90) return "A";
  if (score >= 80) return "B";
  if (score >= 70) return "C";
  if (score >= 60) return "D";
  return "F";
}

// tests
console.log("grade(95):", grade(95));   // A
console.log("grade(85):", grade(85));   // B
console.log("grade(75):", grade(75));   // C
console.log("grade(65):", grade(65));   // D
console.log("grade(55):", grade(55));   // F
console.log("grade(0):", grade(0));     // F
console.log("grade(100):", grade(100)); // A
console.log("grade(-5):", grade(-5));   // null
console.log("grade(101):", grade(101)); // null

File: python/task1b/S1_IF_02.py
# S1_IF_02 - Score to grade

def grade(score):
    if not isinstance(score, (int, float)):
        return None
    if score < 0 or score > 100:
        return None

    if score >= 90:
        return "A"
    elif score >= 80:
        return "B"
    elif score >= 70:
        return "C"
    elif score >= 60:
        return "D"
    else:
        return "F"

# Boundary tests
tests = [59, 60, 69, 70, 79, 80, 89, 90, 100, 101, -1, 0]

print(f"{'score':<8} {'grade'}")
print("-" * 16)
for s in tests:
    print(f"{s:<8} {grade(s)}")
